repos/system_upgrade/el7toel8/actors/peseventsscanner/actor.py

```diff
@@ -56,3 +71,11 @@ def process(self):
         if to_install or to_remove:
             self.produce(RpmTransactionTasks(to_install=list(to_install), to_remove=list(to_remove)))
 
+        for pkg in to_install:
+            repo = pkg_repo.get(pkg, None)
+
+            if repo and repo not in REPOSITORIES_BLACKLIST:
+                to_enable.add(pkg_repo[pkg])
```

Comment:
why are you getting it from dict again, when you already have it in repo variable?